hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SqlDataLinkStatementTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl;
+
+import com.hazelcast.config.DataLinkConfig;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.datalink.DataLink;
+import com.hazelcast.datalink.impl.DataLinkTestUtil.DummyDataLink;
+import com.hazelcast.datalink.impl.InternalDataLinkService;
+import com.hazelcast.jet.sql.SqlTestSupport;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class SqlDataLinkStatementTest extends SqlTestSupport {
```

Comment:
lack of permission tests (should be in EE?) and lack of `information_schema.data_links` tests

Reply:
It's a **separate** PR in EE: https://github.com/hazelcast/hazelcast-enterprise/pull/5785